types/src/v0/impls/state.rs

```diff
@@ -318,17 +318,12 @@ pub fn validate_proposal(
 
     // Check if timestamp is increasing.
     let system_time: u64 = OffsetDateTime::now_utc().unix_timestamp() as u64;
-    if proposal.timestamp() < system_time - 2 {
+    if proposal.timestamp() >= parent_header.timestamp()
+        && proposal.timestamp().abs_diff(system_time) <= 2
```

Comment:
Sorry, this is backwards. This condition is _validity_, but as written we're checking for _in_validity, so we would need to wrap this in `!( )` or flip the inequalities or something

Reply:
Sorry. I'm typing faster than I'm thinking

Reply:
4f474014341baaac1e22af1fe7b3a7e58f4e2c90